### DISCUSSION TITLE
How do i send erc-20 tokens via Metamask?

### DISCUSSION THREAD

**Discussion by di-sukharev** [wevm/wagmi#2510] [Q&A]
Now when i call `sendTransaction(to, amount)` wagmi opens Metamask with chain native token to be sent to `to` address. How do i specify erc-20 contract address, so Metamask is being opened to send non-native, but exactly specified token?

**Comment by SaadNiaziDev** [comment]
Well there is difference between `send()` and `write()` functions. `Send()` functions work on native coins only 😕
You should be using  `contractWrite()` function

**Reply by ghost** [reply]
maybe a stupid question: is there any example of that that uses wagmi `chains`?
I've seen that method and some usages that were very complex and required a lot of manual config (like choosing namespaces, etc)

**Reply by SaadNiaziDev** [reply]
Couldn't understand your question. Can you please simplify that?

**Reply by ghost** [reply]
sorry, I'm new here. I confused wagmi with walletconnect
https://docs.walletconnect.com/2.0/web/sign/dapp-usage

as for wagmi - I read this doc:
https://wagmi.sh/examples/contract-write

and it seems that I need to know the ABI, the address, etc before I do anything.
is there a simpler way to do some very common things?
like sending ERC20 tokens - for example USDC/USDT.
ideally, the same thing would work BEP-20

**Reply by SaadNiaziDev** [reply]
As for ABI, you can use https://wagmi.sh/react/constants/abis#erc-20.
You have to keep address of all the coins in array and choose it dynamically. ABI would be same as you have to target transfer function.

**Reply by ghost** [reply]
fair enough. cheers!

**Comment by engrwaqasali** [comment]
[https://wagmi.sh/react/api/hooks/useWriteContract](https://wagmi.sh/react/api/hooks/useWriteContract
)
You can use the hook useWriteContract

```
const { writeContract } = useWriteContract()

const senToken = (from, to, amount, contractAddress, abi) => {
   writeContract({ 
          abi,
          address: contractAddress ,
          functionName: 'transferFrom',
          args: [
            from,
            to,
           amount,
          ],
       })
 }
```
